lib/membrane_framerate_converter.ex

```diff
@@ -0,0 +1,133 @@
+defmodule Membrane.FramerateConverter do
+  @moduledoc """
+  Element converts video to target constant frame rate, by dropping and duplicating frames as necessary.
+  Input video may have constant or variable frame rate.
+  Element expects each frame to be received in separate buffer.
+  Additionally, presentation timestamps must be passed in each buffer's metadata.
+  """
+
+  use Bunch
+  use Membrane.Filter
+
+  alias Membrane.Buffer
+  alias Membrane.Caps.Video.Raw
+
+  require Membrane.Logger
+
+  def_options framerate: [
+                spec: tuple(),
+                default: {30, 1},
+                description: """
+                Target framerate.
+                """
+              ]
+
+  def_output_pad :output,
+    caps: {Raw, aligned: true}
+
+  def_input_pad :input,
+    caps: {Raw, aligned: true},
+    demand_unit: :buffers
+
+  @impl true
+  def handle_init(%__MODULE__{} = options) do
+    state =
+      options
+      |> Map.from_struct()
+      |> Map.merge(%{
+        last_payload: nil,
+        last_metadata: nil,
+        target_pts: 0
+      })
+
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_demand(:output, size, :buffers, _ctx, state) do
+    {{:ok, demand: {:input, size}}, state}
+  end
+
+  @impl true
+  def handle_process(
+        :input,
+        %{payload: payload, metadata: metadata} = buffer,
+        _ctx,
+        %{last_payload: nil} = state
+      ) do
+    unless Map.has_key?(metadata, :pts), do: raise("cannot cut stream without pts")
+
+    state = %{
+      state
+      | target_pts: metadata.pts,
+        last_metadata: metadata,
+        last_payload: payload
+    }
+
+    {{:ok, [buffer: {:output, buffer}, redemand: :output]}, bump_target_pts(state)}
```

Comment:
1. I'd not hide the function call inside returned value
2. Square brackets around the keyword list can be omitted
```suggestion
    state = %{
      state
      | target_pts: metadata.pts,
        last_metadata: metadata,
        last_payload: payload
    } |> bump_target_pts()

    {{:ok, buffer: {:output, buffer}, redemand: :output}, state}
```